### PR TITLE
Follow-up: chat scrolling refactoring

### PR DESCRIPTION
**Pull request by DorraJaouad**
### ☑️ Resolves

* follow-up to #11934

## TO-DO
- [x] 👷🏻‍♀️ Refactor 
- [x] ⚠️ Fix conversation jumping on opening
- [x] 🔴 Empty content is not shown anymore because of loadedMessages flag does not change to true in empty conv
- [ ] 💚 Addition: add a floating button at the bottom in case you are scrolled up and new messages come in ( aka indicator for new messages while opening the chat in top)

### 🖼️ Screenshots / Screencasts

Conversation switching before

https://github.com/nextcloud/spreed/assets/84044328/480f1393-a9ad-4912-8fde-f2e5248ef616

Conversation switching after

https://github.com/nextcloud/spreed/assets/84044328/34e1e285-68e1-456e-a919-6e8ded5537ef

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [x] 🖥️ Tested with Desktop client or should not be risky for it 
- [x] 🖌️ Design was reviewed, approved or inspired by the design team
- [x] ⛑️ Tests are included or not possible
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

**Comment by DorraJaouad** [comment]
`ChatIdentifier` is updated on two steps:
1. `{token: 't3qj6xtm', isParticipant: false, isInLobby: false}`
Chat is not scrolled yet because , the condition in `handleStartGettingMessagesPreconditions` is faulty
2. `{token: 't3qj6xtm', isParticipant: true, isInLobby: false}`
this is when the chat is finally scrolled

`isParticipant` is not updated instantly, need to find workarounds to make it available

**Comment by DorraJaouad** [comment]
Focused message from URL doesn't focus correctly sometimes 🥲

File: src/components/MessagesList/MessagesList.vue [hunk 1]
```diff
@@ -643,8 +642,7 @@ export default {
 		},
 
 		async handleStartGettingMessagesPreconditions() {
-			if (this.token && this.isParticipant && !this.isInLobby) {
-
+			if (this.token && !this.isInLobby) {
```
Comment (Antreesy) [inline src/components/MessagesList/MessagesList.vue:648]: I would avoid to skip that part, that could lead to some unexpected things.
Though as we knowing some data from conversation, we could do like this (conversation contains the same params as partticipantIdentifier for that conversation, and already known to us):

```js
isParticipant() {
	if (!this.conversation) {
		return false
	}

	return !!this.$store.getters.findParticipant(this.token, this.conversation)
},
```

File: src/store/messagesStore.js [hunk 2]
```diff
@@ -1410,6 +1410,10 @@ const actions = {
 	async easeMessageList(context, { token }) {
 		context.commit('easeMessageList', { token })
 	},
+
+	loadedMessagesOfConversation(context, { token }) {
```
Comment (Antreesy) [inline src/store/messagesStore.js:1414]: Good addition! Please check if we need to clean it up from the store somewhere (like on deleteConversation)

**Review by Antreesy — COMMENTED** [review]
Checked only the last commits (unrelated to scrolling logic):

File: src/store/messagesStore.js (hunk 2 above)
Comment (DorraJaouad) [reply src/store/messagesStore.js:1414]: Nah, it is used to mark the first loading of a conversation so it should be kept there until page reload.

File: src/components/MessagesList/MessagesList.vue (hunk 1 above)
Comment (DorraJaouad) [reply src/components/MessagesList/MessagesList.vue:648]: Perfect 😀

File: src/components/MessagesList/MessagesList.vue [hunk 3]
```diff
@@ -302,8 +301,11 @@ export default {
 					this.softUpdateByDateGroups(this.messagesGroupedByDateByAuthor, newGroups)
 				}
 
-				// scroll to bottom if needed
-				this.scrollToBottom({ smooth: true })
+				// Scroll only if there is no focused messages
+				const focusedMessage = this.getMessageIdFromHash()
```
[the remark this answers is not in the document]
Comment (DorraJaouad) [reply src/components/MessagesList/MessagesList.vue:305]: discussed

File: src/components/MessagesList/MessagesList.vue [hunk 4]
```diff
@@ -613,36 +615,36 @@ export default {
 			let isFocused = null
 			if (focusMessageId) {
 				// scroll to message in URL anchor
-				isFocused = this.focusMessage(focusMessageId, false)
+				return this.focusMessage(focusMessageId)
```
Comment (Antreesy) [inline src/components/MessagesList/MessagesList.vue:618]: returned value would be unexpected here
```suggestion
				this.focusMessage(focusMessageId)
				return
```

File: src/components/MessagesList/MessagesList.vue [hunk 5]
```diff
@@ -653,70 +655,27 @@ export default {
...
-						this.$nextTick(() => {
-							this.scrollToFocusedMessage(focusMessageId)
-						})
-					}
-				}
+				this.$nextTick(() => {
+					// basically scrolling to either the last read message or the message in the URL anchor
+					// and there is a fallback to scroll to the bottom if the message is not found
+					this.scrollToFocusedMessage(focusMessageId)
+				})
```
Comment (Antreesy) [inline src/components/MessagesList/MessagesList.vue:672]: Should prevent unintentional scrollToBottom
```suggestion
					this.isInitialisingMessages = false
				})
```

**Review by Antreesy — APPROVED** [review]
As discussed during pair-review, seems to work much better. Seems that even the opening of every chat at the top is resolved! One more thing to try to reproduce and fix:

**Pull request merged**